internal/core/core.go

```diff
@@ -327,6 +338,12 @@ func Coalesce[T *U, U any](a T, b T) T {
 	}
 }
 
+// Returns the first element that is not `nil`; CoalesceList(a, b, c) is roughly analogous to `a ?? b ?? c` in JS, except that it
+// non-shortcutting, so it is advised to only use a constant or precomputed value for non-first values in the list
+func CoalesceList[T *U, U any](a ...T) T {
```

Comment:
Given how easy it is to misuse `core.IfElse` already for the same reason, I wonder if it's a good idea to have this.

Reply:
I found this useful because expanding chains of || was getting really messy, and there were quite a few `baseSymbol|rootSymbol|originalSymbol|symbol` like statements in find all references, where the list is all precomputed variables. I could change it into just a `findAllReferences` helper function?